src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: T): void => {
+    const valueStr = String(item[key]);
+
+    if (!result[valueStr]) {
+      result[valueStr] = [];
+    }
+
+    result[valueStr].push(item);
+  });
+
+  return result;
```

Comment:
- consider to using method reduce here instead of forEach as it will make your code shorter

Reply:
I've try using reduce but had problem with types...

Reply:
And it wasn't much shorter.